src/processors/AvroSchemaInputProcessor.ts

```diff
@@ -0,0 +1,152 @@
+import { AbstractInputProcessor } from './AbstractInputProcessor';
+import {
+  CommonModel,
+  InputMetaModel,
+  AvroSchema,
+  ProcessorOptions
+} from '../models';
+import { Logger } from '../utils';
+import { Interpreter } from '../interpreter/Interpreter';
+import { convertToMetaModel } from '../helpers';
+
+/**
+ * Class for processing Avro Schema input
+ */
+export class AvroSchemaInputProcessor extends AbstractInputProcessor {
+  /**
+   * Function processing an Avro Schema input
+   *
+   * @param input
+   */
+
+  shouldProcess(input?: any): boolean {
+    if (
+      input === '' ||
+      JSON.stringify(input) === '{}' ||
+      JSON.stringify(input) === '[]'
+    ) {
+      return false;
+    }
+    if (!input.type) {
+      return false;
+    }
+    return true;
+  }
```

Comment:
This function should inspect the input for patterns that form an Avro schema, i.e. figure out if any properties are required in the root object and check for those. If none you can use the optional properties and say "if x amount of properties are found, its avro"

Reply:
As per the [documentation](https://avro.apache.org/docs/1.11.1/specification/#schema-declaration), Avro Schema has a `type` property at the root level, for which I implemented line 30. Moreover, Avro Schema can be defined as a `JSON string`, `JSON Object` or `JSON Array`. Covering all the cases in here.

Reply:
It needs to be more precise because the JSON Schema also has `type` and how it can be defined. 

I would add the following:
- Check for `name` (it might not be there)
- Check the `type` property for null, boolean, int, long, float, double, bytes, string, records, enums, arrays, maps, unions and fixed